packages/kotti-ui/source/kotti-navbar/types.ts

```diff
@@ -17,14 +17,13 @@ export namespace KottiNavbar {
 	})
 	export type QuickLink = z.infer<typeof quickLinkSchema>
 
-	export const sectionLinkSchema = z
-		.object({
-			icon: yocoIconSchema,
-			isActive: z.boolean(),
-			link: z.string().optional(),
-			title: z.string(),
-		})
-		.passthrough()
+	export const sectionLinkSchema = z.object({
+		icon: yocoIconSchema,
+		component: z.string().default('a'),
+		isActive: z.boolean(),
+		linkProps: z.record(z.unknown()).optional(),
```

Comment:
why do you need `record` and not `object`? is this a `zod` limitation you're aware of?

Reply:
I think this is a bit redundant. Any objections to just calling it `props`?

Reply:
I am still considering how we can better restructure this

basically want to combine the `component` and `props` so that you either have `AnchorLink` or a random object

idk if this will be an overkill and actually make us run into more issues
> would have to test myself a bit

but my point is more it's maybe better to either have `a` or something else that can be anything🤷🏼 
> we can even further overkill this with a generic on the component for the linkProps type
so that we end up with 

```
type AnchorLink = `component: 'a' & HTMLLinkElement`
 
sectionLinkSchema<T extends { [key: string]: unknown } = { [key: string]: unknown }> = z.object({
icon: yocoIconSchema,
isActive: z.boolean(), 
type: AnchorLink | T

```

Reply:
@carsoli actually it’s possible to make this conditional via using a `z.array(z.union([z.object(...withA), z.object(...withoutA)]))

However, I don’t think it’s strictly necessary. We also support JS-based links as well which wouldn’t really pass a `href` and perhaps at some point we need other props as well (e.g. `download` to download a zip from the navbar or so).

Another possibility is changing this to `component: { name: 'a', props: { href: '/' } }`, but once again I’m fine with the current solution besides that `linkProps` is a bit redundant

Reply:
I didn't have your answer when I typed the 2nd comment @FlorianWendelborn